### ISSUE TITLE
Components not registering properly with runtime-only vue?

### ISSUE THREAD

**Issue by dts** [vuejs/vue-loader#315]
This may not be the right place for this, and it may be a well-documented feature that I don't know about yet.  That being said, I'm trying to build an app with a vue-router and vue-loader (all for Vue 2.0, with various RCs in there).  If I alias 'vue' to `vue/dist/vue` in my webpack configuration, components work correctly.  However, if I do not, dynamically registered components (like vue-router), don't work in the components coming out of .vue files ( I get a component not registered error ).  Is this a feature, not a bug?  Should I be pointing vue-loader to a list of components I'm going to use?  Did I screw up some configuration somewhere?

Thanks in advance!

**Comment by yyx990803** [comment]
You are expected to use the alias config if you want to use the dist build. Directly importing `vue/dist/vue` without the alias will cause the problems you ran into.

**Comment by dts** [comment]
Is there some documentation somewhere on the runtime version vs. the non-runtime version?  If the runtime-only version doesn't support "globally registered components", how is it supposed to be used?

**Comment by yyx990803** [comment]
There seems to be some confusion here - everything should work using either build, the only difference is the presence of the alias config.

Just make sure you always do `import Vue from 'vue'` in your code.

**Comment by yyx990803** [comment]
https://github.com/vuejs/vue/wiki/Vue-2.0-RC-Starter-Resources#standalone-vs-runtime-builds

**Comment by dts** [comment]
I was having trouble accessing "globally registered" (i.e. by Vue.component() or Vue.use()) components from within *.vue files, and this was resolved when I added that vue alias to the webpack config.

On second thought, I now realize I was including `vue/dist/vue` in my main file, and there must have been other things requiring the `vue` (unaliased).  This caused two vue instances to be active at any time, and so the registrations weren't working properly.

Now that everything is pointing at the same copy of vue, there are no longer two separate registrations happening at the same time...

Sorry about that!
